api/notd/block_processor.py

```diff
@@ -59,14 +76,75 @@ async def get_transfers_in_block(self, blockNumber: int) -> List[RetrievedTokenT
-
-    async def _process_event(self, event: LogReceipt, blockNumber: int, blockHash: str, blockDate: datetime.datetime) -> Optional[RetrievedTokenTransfer]:
+        totalTokenTransferList = []
+        erc721events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc721TansferEventSignatureHash])
+        logging.info(f'Found {len(erc721events)} events in block #{blockNumber}')
+        for erc721EventsChunk in list_util.generate_chunks(erc721events, 5):
+            totalTokenTransferList += [tokenTransfer for tokenTransfer in await asyncio.gather(*[self._process_erc721_single_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc721EventsChunk]) for tokenTransfer in tokenTransfer]
+
+        erc1155events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc1155TansferEventSignatureHash])
+        logging.info(f'Found {len(erc1155events)} erc1155SingleEvents in block #{blockNumber}')
+        for erc1155EventsChunk in list_util.generate_chunks(erc1155events, 5):
+            totalTokenTransferList += [tokenTransfer for tokenTransfer in await asyncio.gather(*[self._process_erc1155_single_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc1155EventsChunk]) for tokenTransfer in tokenTransfer]
+
+        erc1155Batchevents = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc1155TansferBatchEventSignatureHash])
+        logging.info(f'Found {len(erc1155Batchevents)} erc1155BatchEvents in block #{blockNumber}')
+        for erc1155BatchEventsChunk in list_util.generate_chunks(erc1155Batchevents, 5):
+            totalTokenTransferList += [tokenTransfer for tokenTransfer in await asyncio.gather(*[self._process_erc1155_batch_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc1155BatchEventsChunk]) for tokenTransfer in tokenTransfer]
+        return  totalTokenTransferList
+
+    async def _process_erc1155_single_event(self, event: LogReceipt, blockNumber: int, blockHash: str, blockDate: datetime.datetime) -> Optional[List[RetrievedTokenTransfer]]:
+        transactionHash = event['transactionHash'].hex()
+        registryAddress = event['address']
+        logging.debug(f'------------- {transactionHash} ------------')
+        if len(event['topics']) < 4:
+            logging.debug('Ignoring event with less than 4 topics')
+            return []
+        operatorAddress = normalize_address(event['topics'][1].hex())
+        fromAddress = normalize_address(event['topics'][2].hex())
+        toAddress = normalize_address(event['topics'][3].hex())
+        data = event['data']
+        data = textwrap.wrap(data[2:], 64)
+        data = [literal_eval(f'0x{elem}') for elem in data]
```

Comment:
what is literal_eval doing here? kinda seems like it might not be needed

Reply:
using it to convert hex to -> int , the array count, token ids and values are stored has hexs

Reply:
i dont understand if you remove the eval what breaks below?

Reply:
length of Array becomes Hex
tokenIDs become hex and amount remain in hex

Reply:
ahh. is there not a better way to convert hex to string? maybe something in like web3 util. literal_eval is a super dangerous function so i would def prefer we avoid it if possible